src/bloomd/networking.c

```
@@ -219,12 +224,54 @@ static int setup_tcp_listener(bloom_networking *netconf) {
     }
 
     // Create the libev objects
-    ev_io_init(&netconf->tcp_client, handle_new_client,
+    ev_io_init(&netconf->tcp_client, handle_new_tcp_client,
                 tcp_listener_fd, EV_READ);
     ev_io_start(netconf->default_loop, &netconf->tcp_client);
     return 0;
 }
 
+/**
+ * Initializes the Unix domain socket listener 
+ * @arg netconf The network configuration
+ * @return 0 on success.
+ */
+static int setup_unix_listener(bloom_networking *netconf) {
+    struct sockaddr_un addr;
+    int unix_listener_fd;
+
+    bzero(&addr, sizeof(addr));
+    addr.sun_family = AF_UNIX;
+
+    unlink(netconf->config->unix_socket);
+
+    strncpy(addr.sun_path, netconf->config->unix_socket, sizeof(addr.sun_path) - 1);
+
+    if ((unix_listener_fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1) {
+        syslog(LOG_ERR, "Failed create UNIX socket! Err: %s", strerror(errno));
+        return 1;
+    }
+    if (bind(unix_listener_fd, (struct sockaddr*)&addr, sizeof(addr)) != 0) {
+        syslog(LOG_ERR, "Failed to bind on UNIX socket! Err: %s", strerror(errno));
+        close(unix_listener_fd);
+        return 1;
+    }
+    if (listen(unix_listener_fd, BACKLOG_SIZE) != 0) {
+        syslog(LOG_ERR, "Failed to listen on UNIX socket! Err: %s", strerror(errno));
+        close(unix_listener_fd);
+        return 1;
+    }
+
+    if (netconf->config->unix_socket_perm) {
+        chmod(addr.sun_path, netconf->config->unix_socket_perm);
```

better check for error, chmod should return -1 on error. 
I'm also not sure that `strtol` works correctly here, so there is chance that chmod will silently fail